cf-oscore/src/main/java/org/eclipse/californium/oscore/HashMapCtxDB.java

```diff
@@ -341,7 +344,12 @@ private static String normalizeServerUri(String uri) throws OSException {
 			LOGGER.error("Error finding host of request URI: " + uri + " message: " + e.getMessage());
 		}
 		if (ipv6Addr instanceof Inet6Address) {
-			normalized = ipv6Addr.getHostAddress();
+			normalized = "[" + ipv6Addr.getHostAddress() + "]";
+		}
+
+		// Consider port, if not default
+		if (port != -1 && port != CoAP.DEFAULT_COAP_PORT) {
```

Comment:
Just as idea, if port is initialized above with `CoAP.DEFAULT_COAP_PORT`, then the `-1` check may be left out.

Reply:
Yeah, true I can modify things like that and it should be fine.

Reply:
Actually, upon testing this it doesn't work since this line:
`port = (new URI(uri)).getPort();`
Will assign -1 to port if the uri doesn't contain a port. So I still need the -1 check in the end.